### PR TITLE
[8.0.x] JBPM-5531: Problems after deleting a form that is used as a subform in other form.

### PR DESCRIPTION
**Pull request by pefernan**
Copy of https://github.com/kiegroup/optaplanner-wb/pull/192

There are some services changed on DataModeller that were used here... @mcimbora @jsoltes can you review it?

This PR is part of an ensemble, please merge with:
https://github.com/kiegroup/droolsjbpm-build-bootstrap/pull/493
https://github.com/kiegroup/kie-wb-common/pull/1011
https://github.com/kiegroup/drools-wb/pull/564
https://github.com/kiegroup/optaplanner-wb/pull/193
https://github.com/kiegroup/jbpm-wb/pull/819
https://github.com/kiegroup/kie-wb-distributions/pull/565
https://github.com/kiegroup/jbpm-form-modeler/pull/126

### CODE REVIEW

File: optaplanner-wb-screens/optaplanner-wb-domain-editor/optaplanner-wb-domain-editor-backend/src/main/java/org/optaplanner/workbench/screens/domaineditor/backend/server/validation/PlanningSolutionSaveValidator.java [hunk 1]
```diff
@@ -27,10 +27,13 @@
 import org.guvnor.common.services.shared.validation.model.ValidationMessage;
 import org.kie.workbench.common.screens.datamodeller.service.DataModelerService;
 import org.kie.workbench.common.services.datamodeller.core.DataObject;
+import org.kie.workbench.common.services.refactoring.service.AssetsUsageService;
+import org.kie.workbench.common.services.refactoring.service.ResourceType;
 import org.kie.workbench.common.services.shared.validation.SaveValidator;
 import org.optaplanner.workbench.screens.domaineditor.validation.PlanningSolutionToBeDuplicatedMessage;
 import org.uberfire.backend.vfs.Path;
 
+import static org.optaplanner.workbench.screens.domaineditor.model.PlannerDomainAnnotations.PLANNING_ENTITY_ANNOTATION;
```
Comment (mcimbora) [inline optaplanner-wb-screens/optaplanner-wb-domain-editor/optaplanner-wb-domain-editor-backend/src/main/java/org/optaplanner/workbench/screens/domaineditor/backend/server/validation/PlanningSolutionSaveValidator.java]: Looks like this one is unused.

**Review by mcimbora — COMMENTED** [review]
Looks good to me, just a minor comment regarding unused import.

**Comment by pefernan** [comment]
@mcimbora thanks! Better now?

**Review by mcimbora — APPROVED** [review]

**Comment by mcimbora** [comment]
@pefernan Looks good, thanks.

**Comment by manstis** [comment]
@jsoltes You haven't approved this one.

**Comment by manstis** [comment]
@jsoltes Could you please approve this?

**Review by jsoltes — APPROVED** [review]
@manstis I am sorry I wasn't receiving emails for this repository. All fixed now.

**Pull request merged**